scripts/mbedtls_framework/build_tree.py

```diff
@@ -7,17 +7,24 @@
 
 import os
 import inspect
+import re
 from typing import Optional
 
 def looks_like_tf_psa_crypto_root(path: str) -> bool:
     """Whether the given directory looks like the root of the PSA Crypto source tree."""
-    return all(os.path.isdir(os.path.join(path, subdir))
-               for subdir in ['include', 'core', 'drivers', 'programs', 'tests'])
+    try:
+        with open(os.path.join(path, 'scripts', 'project_name.txt'), 'r') as f:
+            return f.read() == "TF-PSA-Crypto\n"
+    except FileNotFoundError:
+        return False
 
 def looks_like_mbedtls_root(path: str) -> bool:
     """Whether the given directory looks like the root of the Mbed TLS source tree."""
-    return all(os.path.isdir(os.path.join(path, subdir))
-               for subdir in ['include', 'library', 'programs', 'tests'])
+    try:
+        with open(os.path.join(path, 'scripts', 'project_name.txt'), 'r') as f:
+            return f.read() == "Mbed TLS\n"
```

Comment:
It feels a bit redudant to have two identical methods for two different comparison operators. Also the name could be confusing since we are looking for the project rather than the root directory.

I would propose moving forward to have a uniform method:

    get_project_from_path(path:str)-> str: 

That returns the name of the project, or "" if invalid.

Reply:
Yes we have twice the same `open` line of code but only twice thus I do not feel the need to factorize it in a dedicated function.
Otherwise I think the name of the function are still fine, they still detect if `path` looks like the root of Mbed TLS or TF-PSA-Crypto, just they do it differently.